crates/telio-wg/src/link_detection.rs

```diff
@@ -0,0 +1,477 @@
+    }
+
+    #[test]
+    fn test_is_disabled_config() {
+        // Disabled
+        let no_link_detection = LinkDetection::new(None);
+        assert_eq!(no_link_detection.is_disabled(), true);
+
+        // Enabled
+        let features = FeatureLinkDetection { rtt_seconds: 1 };
+        let no_link_detection = LinkDetection::new(Some(features));
+        assert_eq!(no_link_detection.is_disabled(), false);
+    }
+
+    #[tokio::test(start_paused = true)]
+    async fn test_time_since_last_rx_with_disabled_link_detection() {
+        let mut ld = LinkDetectionDisabled::default();
+        let key = PublicKey::default();
+
+        assert_eq!(ld.time_since_last_rx(&key), None);
+
+        // Insert
+        ld.insert(&key, None);
+        assert_eq!(ld.time_since_last_rx(&key), Some(Instant::now().elapsed()));
+        ld.remove(&key);
+
+        time::advance(Duration::from_secs(1)).await;
+
+        // Insert with value
+        ld.insert(&key, Some(5));
+        let last_change = Instant::now();
+        assert_eq!(ld.time_since_last_rx(&key), Some(last_change.elapsed()));
+
+        time::advance(Duration::from_secs(1)).await;
+
+        // Update with no change in rx_bytes, so there will be no timestamp update
+        ld.update(&key, Some(5), false);
+        assert_eq!(ld.time_since_last_rx(&key), Some(last_change.elapsed()));
+
+        time::advance(Duration::from_secs(1)).await;
+
+        // Update with different rx_bytes
+        ld.update(&key, Some(10), false);
+        assert_eq!(ld.time_since_last_rx(&key), Some(Instant::now().elapsed()));
+    }
+
+    #[tokio::test(start_paused = true)]
+    async fn test_time_since_last_rx_with_enabled_link_detection() {
+        let mut ld = LinkDetectionEnabled::new(Duration::from_secs(15));
+        let key = PublicKey::default();
+
+        assert_eq!(ld.time_since_last_rx(&key), None);
+
+        // Insert
+        ld.insert(&key, None, None, LinkState::Down);
+        assert_eq!(ld.time_since_last_rx(&key), Some(Instant::now().elapsed()));
+        ld.remove(&key);
+
+        time::advance(Duration::from_secs(1)).await;
+
+        // Insert with value
+        ld.insert(&key, Some(5), None, LinkState::Down);
+        let last_change = Instant::now();
+        assert_eq!(ld.time_since_last_rx(&key), Some(last_change.elapsed()));
+
+        time::advance(Duration::from_secs(1)).await;
+
+        // Update with no change in rx_bytes, so there will be no timestamp update
+        ld.update(&key, Some(5), None, NodeState::Connected, false);
+        assert_eq!(ld.time_since_last_rx(&key), Some(last_change.elapsed()));
+
+        time::advance(Duration::from_secs(1)).await;
+
+        // Update with different rx_bytes
+        ld.update(&key, Some(10), None, NodeState::Connected, false);
+        assert_eq!(ld.time_since_last_rx(&key), Some(Instant::now().elapsed()));
+    }
+
+    #[tokio::test(start_paused = true)]
+    async fn test_compute_new_node_info_with_no_link_state_updates() {
```

Comment:
This test function doesn't do anything async so it can be a standard `#[test]`. I also think it makes sense to define `check_node_info` inside this fn since it's the only place it's called, and I think if you define it as a closure instead of a function and do that after creating `old`, you don't even need to pass old to the closure. That explanation might be a bit unclear so:
```
let name = "telio";
let speak = || println!("Hello, {name}!");
speak();
```

Reply:
The test is async because I need time to be paused, If you look down I assert with Instant::now().
I moved check_node_info into a closure. I will think for a better solution in the next PR.

Reply:
My bad about the async!